ventilator_database.py

```diff
@@ -29,30 +29,60 @@ def store_trigger(self, trigger_val):
         collection = self.db.trigger_values
         self.__store_value(collection, trigger_val)
 
-
     def __store_value(self, collection, val):
         try:
             collection.insert_one({'value': val, 'loggedAt': datetime.utcnow()})
         except errors.ConnectionFailure:
             print("Lost connection, client will attempt to reconnect")
 
+    def last_n_data(self, type_data, N=100):
+        # retrieve the last "N" added measurement 
+        if type_data == 'BPM':
+            return self.db.breathsperminute_values.find().sort("loggedAt", -1).limit(N)
```

Comment:
Wouldn't it make more sense to keep a circular buffer of these values locally, rather than having to go out to the database latency wise?